source/controlTypes.py

```diff
@@ -622,27 +623,44 @@
 	ROLE_APPLICATION,
 }
 
-#{ Output reasons
-# These constants are used to specify the reason that a given piece of output was generated.
-#: An object to be reported due to a focus change or similar.
-REASON_FOCUS="focus"
-#: An ancestor of the focus object to be reported due to a focus change or similar.
-REASON_FOCUSENTERED="focusEntered"
-#: An item under the mouse.
-REASON_MOUSE="mouse"
-#: A response to a user query.
-REASON_QUERY="query"
-#: Reporting a change to an object.
-REASON_CHANGE="change"
-#: A generic, screen reader specific message.
-REASON_MESSAGE="message"
-#: Text reported as part of a say all.
-REASON_SAYALL="sayAll"
-#: Content reported due to caret movement or similar.
-REASON_CARET="caret"
-#: No output, but any state should be cached as if output had occurred.
-REASON_ONLYCACHE="onlyCache"
-#}
+
+class OutputReason(Enum):
+	"""Specify the reason that a given piece of output was generated.
+	"""
+	#: An object to be reported due to a focus change or similar.
+	FOCUS = auto()
+	#: An ancestor of the focus object to be reported due to a focus change or similar.
+	FOCUSENTERED = auto()
+	#: An item under the mouse.
+	MOUSE = auto()
+	#: A response to a user query.
+	QUERY = auto()
+	#: Reporting a change to an object.
+	CHANGE = auto()
+	#: A generic, screen reader specific message.
+	MESSAGE = auto()
+	#: Text reported as part of a say all.
+	SAYALL = auto()
+	#: Content reported due to caret movement or similar.
+	CARET = auto()
+	#: No output, but any state should be cached as if output had occurred.
+	ONLYCACHE = auto()
+
+	QUICKNAV = auto()
+
+# The following constants are kept for backwards compatibility.
+# In future, OutputReason should be used directly
+
+
+REASON_FOCUS = OutputReason.FOCUS

```

Comment:
As there are several other places we should use enums in future (textInfos.position, textInfos.unit,  controlTypes.role, controlTypes.state), I'd like to see the exploding of the enum values into the outer scope here be more generic rather than explicitly listing each one, as it would be very easy to forget one.
Something like:
```
for r in OutputReason:
	globals()['REASON_%s' % r.name] = r
```

Reply:
One thing I don't like about doing this, is that it means new constants also become available automatically at this scope. Forgetting to update this "exploded enum" when adding new constants is a feature in my mind. I agree it means when creating the enum and exploded values we have to be careful to double check that they are all available, but just like the `quickNav` value it is easy to miss one because it is defined in another file. I want to deprecate the constants at this scope and remove them in 2021.1

Reply:
Fair enough. I'm not sure how I'd feel about a new value being added to the enum in the future and not being then duplicated at module level, unless we go through the code and remove all model-level references to the values. I think defining some differently is extremely confusing. I certainly agree that values for new enums should never be duplicated, but pre-existing ones should either be duplicated and or all module-level references removed.
However, I'll approve this as is, and you can make the final decision.

Reply:
Since this stemmed from the speech function refactor I only did "the minimal". But I think we should follow up with another PR to do the same for the rest of the constants in control types, and ensure that all of NVDA uses the enum versions and officially deprecate (and give a date of removal for) the module level constants.

Reply:
I have created #10732 to do this follow up work.

Reply:
I have also created a new label `deprecated/2021.1` to keep track of deprecations and their removal date. This should make it easier to give a concrete set of "breaking changes" in the 2021.1 release.